Entities/Common/Scripts/DecayStuckArrows.as

```diff
@@ -0,0 +1,50 @@
+Random rand(Time());
+
+void onTick(CBlob@ this)
+{
+	string[]@ names;
+	uint[]@ times;
+
+	if (this.get("stuck_arrow_names", @names) && this.get("stuck_arrow_times", @times))
+	{
+		CSprite@ sprite = this.getSprite();
+
+		for (int i = 0; i < times.size(); i++)
+		{
+			string name = names[i];
+			uint time = times[i];
+			bool decay = time <= getGameTime();
+			CSpriteLayer@ arrow = sprite.getSpriteLayer(name);
+
+			//remove if arrow should decay, blob is dead, or player turned on fast render
+			if (decay || this.hasTag("dead") || v_fastrender)
+			{
+				//only show gibs on decay
+				if (decay)
```

Comment:
I think you should remove the decay check before making particles.

When a player dies, the arrows just disappear from it, which is a bit odd, making it do a particle effect makes it look much better imo

Reply:
Adding to this, maybe only do the particle gib if its on the players screen